app/src/test/java/org/openobservatory/ooniprobe/common/PreferenceManagerTest.java

```diff
@@ -0,0 +1,471 @@
+package org.openobservatory.ooniprobe.common;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.text.format.DateFormat;
+
+import org.junit.Test;
+import org.openobservatory.engine.OONISession;
+import org.openobservatory.ooniprobe.R;
+import org.openobservatory.ooniprobe.RobolectricAbstractTest;
+import org.openobservatory.ooniprobe.engine.TestEngineInterface;
+import org.openobservatory.ooniprobe.test.EngineProvider;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import static androidx.preference.PreferenceManager.getDefaultSharedPreferences;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.openobservatory.ooniprobe.common.PreferenceManager.AUTORUN_COUNT;
+import static org.openobservatory.ooniprobe.common.PreferenceManager.AUTORUN_DATE;
+import static org.openobservatory.ooniprobe.common.PreferenceManager.DELETE_JSON_DELAY;
+import static org.openobservatory.ooniprobe.common.PreferenceManager.DELETE_JSON_KEY;
+import static org.openobservatory.ooniprobe.common.PreferenceManager.NOTIFICATION_DIALOG_DISABLE;
+import static org.openobservatory.ooniprobe.common.PreferenceManager.SHOW_ONBOARDING;
+import static org.openobservatory.ooniprobe.common.PreferenceManager.TOKEN;
+import static org.openobservatory.ooniprobe.common.PreferenceManager.UUID4;
+
+public class PreferenceManagerTest extends RobolectricAbstractTest {
+
+    @Test
+    public void testGetSetToken() {
+        // Arrange
+        PreferenceManager pm = build(c);
+        SharedPreferences sharedPreferences = getDefaultSharedPreferences(c);
```

Comment:
You either use a different sharedPreferences instance, or you need to clear it after use.